### ISSUE TITLE
Can't resolve vite aliases since 0.5.0

### ISSUE THREAD

**Issue by francois-codes** [Valian/live_vue#40]
I was on 0.4.1 and I migrated to 0.5.5

this change
```markdown

- `assets/vue/index.js` should export app created by `createLiveVue()`, not just available components. See migration below.

```

seems to break the resolution of components imported with an alias set in the vite config. 

I will try to spend more time to figure out why that doesn't work anymore, but it probably has to do with the new resolution mechanism with `findComponent`

for now, I'll stick to 0.4.1

**Comment by Valian** [comment]
It's hard for me to help you without more details @francois-codes.

Could you show me your index.js file before / after migration? Also, vite config and exact error you're getting would be extremely helpful!

`findComponent` was in 0.4.1 as well, just hidden and not visible to the user. The only role of that function is to take name of the component passed from backend, take all the available components and their paths, and return resolved component. FYI that's the implementation:

```javascript
  // we're looking for a component by name or path suffix.
  for (const [key, value] of Object.entries(components)) {
    if (key.endsWith(`${name}.vue`) || key.endsWith(`${name}/index.vue`)) {
      return value
    }
  }
```

**Comment by francois-codes** [comment]
Hi Valian, 

Yes I'm sorry for the short report, and the nasty tone, which wasn't intended. I was pretty busy and couldn't afford to allocate more time to debug this at the time, but still wanted to report the issue - in case I missed something obvious or that was a known issue. I assumed too quickly that since you moved to typescript and refactored the app's init, it would have been easy to mess up something with the component path resolution by using the wrong module resolution or something.

I spent more time on it today and it turned out it was nothing like this. Just a vite cache issue - not sure why, one of my assumption is that since the package is sym-linked from deps in node_modules, vite missed an update or something... not sure,  but after starting the vite server with `--force` that resolution error went away.

I did come across another issue though - which may not require a change in the code but would be worth referencing somewhere as I can easily see some people running through this issue.

Basically I copied the code from the changelog for the app initialisation, which includes `{ eager: true }` in the import statements for resolving components.  I noticed that this caused an expected issue: it was evaluating some of my components before evaluating the LiveView hook which calls `createLiveVue().setup` function. As a result, the app crashed on launch as it was trying to use a pinia store that wasn't initialised (since `setup` hadn't been called yet)

So it looks like you have to use `{ eager: false }` when importing components if you want to be sure your Vue app's setup function runs before your components are evaluated. Happy to push a PR to make a note of that in the README or somewhere if you think that's relevant.

Thanks for the very useful lib !

**Comment by Valian** [comment]
> Yes I'm sorry for the short report, and the nasty tone, which wasn't intended

No problem, didn't noticed it ;) Simply needed more details to try to aid you. 

Ok so there's a few things to unpack:

1. `{eager: true}` only changes the way component is imported and if separate bundles are created for each component. With `eager: true`, Vite produces code that simply imports each found file immediately, at the top of the file:

```js
const modules = import.meta.glob('./dir/*.js', {
  import: 'default',
  eager: true,
})
```

```js
// code produced by vite:
import __glob__0_0 from './dir/foo.js'
import __glob__0_1 from './dir/bar.js'
const modules = {
  './dir/foo.js': __glob__0_0,
  './dir/bar.js': __glob__0_1,
}
```

With eager true, these imports are "lazy". See more [here](https://vite.dev/guide/features#glob-import).

2. Importing component doesn't call "setup", so a place where you most likely initialize Pinia stores. Setup is called only when component is rendered.

3. In the new syntax introduced in v0.5.0, you should move your pinia initialization to `createLiveVue().setup()`. Is it how you've done it? An example:

```js
  setup: ({ createApp, component, props, slots, plugin, el }) => {
    const app = createApp({ render: () => h(component, props, slots) })
    app.use(plugin)
    // add your own plugins here
    app.use(pinia)
    app.mount(el)
    return app
  },
```

The lifecycle of the LiveVue instance is as follows:

1. HTML is rendered by LiveView
2. LiveVue hook mounted() function is called by LiveView
3. We resolve which component should be rendered using `createLiveVue().resolve(name)` which you implemented
4. We create a Vue app using `createLiveVue().setup(context)` which again you provided. It also mounts the vue app to the DOM.

And that's it. Not sure where there might be races 🤔 Would be great if you could provide relevant snippets - how you initialize pinia, how your `assets/vue/index.js` looks like.

**Comment by francois-codes** [comment]
Yes, my app is initialized like you describe, I have implemented my vue init logic (including setting pinia stores) in the setup function inside `createLiveVue()`

In my app, I have components using stores like this

```vue
<script setup lang="ts'>

const store = useSomePiniaStore()

</script>

<template>
 // ...
</template>
```

somehow, if I declare the resolve function like this
```
createLiveVue({
  resolve: name => {
    const components = {
      ...import.meta.glob("./**/*.vue", { eager: true }),
      ...import.meta.glob("../../lib/**/*.vue", { eager: true }),
  }
})
```

Then I get the missing pinia store error before I can even break into the invocation of the liveVueHook. If I remove the eager load, or if I provide a map of dynamic imports, then it's fine. I can't really explain it either 🤷🏻‍♂️ , maybe something happening on the vite side, but in any case, that could be a hurdle.

I'm not using SSR, haven't tried if that makes a difference.

I'll try to push a repo that reproduces the problem so you can see (and I can check that I'm not crazy 🙃 )

**Comment by francois-codes** [comment]
@Valian I'm closing this as the other issue I encountered is different. I'll create a new issue with a reproduction repo when I get some time to get that going
